superbench/benchmarks/micro_benchmarks/ib_validation_performance.py

```diff
@@ -190,21 +205,26 @@ def __fully_one_to_one(self, n):
             candidates = non_moving + robin
         return config
 
-    def gen_traffic_pattern(self, n, mode, config_file_path):
+    def gen_traffic_pattern(self, hosts, mode, ibstat, ibnetdiscover, config_file_path):
         """Generate traffic pattern into config file.
 
         Args:
-            n (int): the number of nodes.
+            hosts (list): the list of hosts read from hostfile.
             mode (str): the traffic mode, including 'one-to-one', 'one-to-many', 'many-to-one'.
+            ibstat (str): the path of ibstat output.
+            ibnetdiscover (str): the path of ibnetdiscover output.
             config_file_path (str): the path of config file to generate.
         """
         config = []
+        n = len(hosts)
         if mode == 'one-to-many':
             config = self.__one_to_many(n)
         elif mode == 'many-to-one':
             config = self.__many_to_one(n)
         elif mode == 'one-to-one':
             config = self.__fully_one_to_one(n)
+        elif mode == 'topo-aware' and ibstat.strip() and ibnetdiscover.strip():
```

Comment:
default `ibstat` and `ibnetdiscover` is `None`, `.strip()` won't work in that case?

Reply:
shall we check `ibstat` and `ibnetdiscover` inside `gen_topo_aware_config`? I think None is one case to handle inside gen_topo_aware_config.

Reply:
Moved the check inside `gen_topo_aware_config` in new revision.